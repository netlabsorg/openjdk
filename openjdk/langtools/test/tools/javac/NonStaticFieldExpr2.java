/* @test  /nodynamiccopyright/
   @bug 4087127 4785453
   @author dps
   @summary method: instance access through types is not allowed

   @run shell NonStaticFieldExpr2.sh
*/

class NonStaticFieldExpr2 {

  public int x;

  void foo () {
     int z = NonStaticFieldExpr2.x;     // SHOULD BE ERROR
  }
}
