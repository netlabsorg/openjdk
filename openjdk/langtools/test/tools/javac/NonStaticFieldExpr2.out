NonStaticFieldExpr2.java:14: non-static variable x cannot be referenced from a static context
     int z = NonStaticFieldExpr2.x;     // SHOULD BE ERROR
                                ^
1 error
